pyop2/base.py

```diff
@@ -3640,17 +3657,22 @@ def name(self):
         """Kernel name, must match the kernel function name in the code."""
         return self._name
 
-    @property
```

Comment:
The pattern you are using here is implemented as `utils.cached_property` in Firedrake. I think it would be clearer to just use that.